opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScan.java

```diff
@@ -126,6 +110,54 @@ public void close() {
 
   @Override
   public String explain() {
-    return getRequestBuilder().build().toString();
+    return request.toString();
+  }
+
+  /** No-args constructor.
+   * @deprecated Exists only to satisfy Java serialization API.
+   */
+  @Deprecated(since = "introduction")
+  public OpenSearchIndexScan() {
+  }
+
+  @Override
+  public void readExternal(ObjectInput in) throws IOException {
+    int reqSize = in.readInt();
+    byte[] requestStream = new byte[reqSize];
+    in.read(requestStream);
+
+
+    var engine = (OpenSearchStorageEngine) ((PlanSerializer.CursorDeserializationStream) in)
+        .resolveObject("engine");
+
+    try (BytesStreamInput bsi = new BytesStreamInput(requestStream)) {
+
+      request = new OpenSearchScrollRequest(bsi, engine);
+    }
+    maxResponseSize = in.readInt();
+
+    client = engine.getClient();
+
+  }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    if (!request.hasAnotherBatch()) {
+      throw new NoCursorException();
+    }
+
+    BytesStreamOutput reqOut = new BytesStreamOutput();
+    request.writeTo(reqOut);
+    reqOut.flush();
+
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    reqOut.bytes().writeTo(baos);
+    baos.flush();
+
+    var reqAsBytes = baos.toByteArray();
+    out.writeInt(reqAsBytes.length);
+    out.write(reqAsBytes);
```

Comment:
Why intermediate streams are required? May be leave a comment

Reply:
Fair point. At the time didn't see a better way to serialize request to byte stream.

Found a better way and commented this section.